src/IRrecv.cpp

```diff
@@ -242,8 +243,20 @@ static void USE_IRAM_ATTR gpio_intr() {
   // @see https://github.com/espressif/arduino-esp32/blob/6b0114366baf986c155e8173ab7c22bc0c5fcedc/cores/esp32/esp32-hal-timer.c#L176-L178
   timer->dev->config.alarm_en = 1;
 #else  // _ESP32_IRRECV_TIMER_HACK
-  timerWrite(timer, 0);
-  timerAlarmEnable(timer);
+  // Check the ESP32 core version
+#if defined(ESP_ARDUINO_VERSION) && \
```

Comment:
Can `#elif` be used here to avoid indentation and the extra `#endif`

What I'm looking for is a diff that only has added rows if possible.

I would also suggest to split out the `ARDUINO_VERSION` check to a new `ARDUINO_COREV3` that can be reused. And that way reduce the repeats.

If we have only additions with COREV3 conditionals, and no removed code lines in the diff, we can be sure nothing gets broken and thus safely merge this.